### PR TITLE
22.09 release updates

### PR DESCRIPTION
**Pull request by ericenns**
## Description of changes
Bump version number in preparation for release and add upgrading notes for API OAuth2 changes.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [ ] Tests added (or description of how to test) for any new features.
* [ ] User documentation updated for UI or technical changes.

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -4,6 +4,20 @@ Upgrading
 This document summarizes the environmental changes that need to be made when
 upgrading IRIDA that cannot be automated.
 
+22.05 to 22.09
+--------------
+* This upgrade switches the OAuth2 implementation from using spring-security-oauth to spring-security-oauth2-authorization-server and spring-security-oauth2-resource-server. Every administrator will need to update their deployments to configure a java keystore used for encrypting OAuth2 access tokens. This java keystore will need to be present on all servers which allow api access. Otherwise a token generated on one server will not work on the other server.
+  * The following is an example command that can be run to generate an appropriate java keystore:
```
Comment (apetkau) [inline UPGRADING.md:10]: Maybe just mention that `keytool` is a command that should come with your Java installation to save people having to lookup how to install `keytool`?

File: UPGRADING.md [hunk 2]
```diff
@@ -4,6 +4,20 @@ Upgrading
...
+  ```bash
+  keytool -genkeypair -alias JWK -keyalg RSA -noprompt -dname CN=irida.bioinformatics.corefacility.ca, OU=ID, O=IRIDA, L=IRIDA, S=IRIDA, C=CA -keystore /etc/irida/jwk-key-store.jks -validity 3650 -storepass SECRET -keypass SECRET -storetype PKCS12
```
Comment (apetkau) [inline UPGRADING.md:12]: I believe `CN=irida.bioinformatics.corefacility.ca, OU=ID, O=IRIDA, L=IRIDA, S=IRIDA, C=CA` needs to be in quotes.

Also, could could you maybe mention that this is just information used to identify the key with a particular organization and that people can fill out whatever they want here (or just leave out the `-dname` parameter and `keytool` will prompt you for the information). That way at least people will not mistakenly think they need to copy/paste the exact string we've used here.

File: UPGRADING.md [hunk 3]
```diff
@@ -4,6 +4,20 @@ Upgrading
 This document summarizes the environmental changes that need to be made when
 upgrading IRIDA that cannot be automated.
 
+22.05 to 22.09
+--------------
+* This upgrade switches the OAuth2 implementation from using spring-security-oauth to spring-security-oauth2-authorization-server and spring-security-oauth2-resource-server. Every administrator will need to update their deployments to configure a java keystore used for encrypting OAuth2 access tokens. This java keystore will need to be present on all servers which allow api access. Otherwise a token generated on one server will not work on the other server.
```
Comment (apetkau) [inline UPGRADING.md:9]: Could you mention that encrypting and transferring OAuth2 access tokens is done using JSON Web Tokens and maybe link to some more information (e.g., https://jwt.io/introduction). That way people have more context as to why they need to setup a Java keystore.

File: UPGRADING.md [hunk 4]
```diff
@@ -4,6 +4,20 @@ Upgrading
...
+  * The following is an example command that can be run to generate an appropriate java keystore:
+  ```bash
+  keytool -genkeypair -alias JWK -keyalg RSA -noprompt -dname CN=irida.bioinformatics.corefacility.ca, OU=ID, O=IRIDA, L=IRIDA, S=IRIDA, C=CA -keystore /etc/irida/jwk-key-store.jks -validity 3650 -storepass SECRET -keypass SECRET -storetype PKCS12
+  ```
+  * After generating the java keystore as above you will need to update `/etc/irida/irida.conf` to specify the keystore and its password:
+  ```text
+  oauth2.jwk.key-store=file:/etc/irida/jwk-key-store.jks
```
Comment (apetkau) [inline UPGRADING.md:16]: Do you need `file:` here? I thought I used it just with the path to the keystore and it worked. But maybe both work.

**Review by apetkau — CHANGES_REQUESTED** [review]
Hey Eric, this looks great. I just have a few suggestions.

File: UPGRADING.md (hunk 4 above)
Comment (ericenns) [reply UPGRADING.md]: Removed in 84a40d49, it isn't required I am just used to including it

File: UPGRADING.md (hunk 3 above)
Comment (ericenns) [reply UPGRADING.md]: Added in  84a40d9

File: UPGRADING.md (hunk 2 above)
Comment (ericenns) [reply UPGRADING.md]: Added in  84a40d9

File: UPGRADING.md (hunk 1 above)
Comment (ericenns) [reply UPGRADING.md]: added in  84a40d9

**Review by apetkau — CHANGES_REQUESTED** [review]
Thanks for all the changes. Just one minor thing to fix.

File: UPGRADING.md [hunk 5]
```diff
@@ -4,6 +4,20 @@ Upgrading
 This document summarizes the environmental changes that need to be made when
 upgrading IRIDA that cannot be automated.
 
+22.05 to 22.09
+--------------
+* This upgrade switches the OAuth2 implementation from using spring-security-oauth to spring-security-oauth2-authorization-server and spring-security-oauth2-resource-server. Due to the dependency updates we have changed the format of the OAuth2 access tokens, they are now JWT Tokens(https://jwt.io/introduction) and are encrypted/decrypted using a certificate within a java keystore. No default java keystore is provided, so administrators will need to update their deployments to configure an appropriate java keystore. The same java keystore will need to be present on all servers which allow api access, otherwise access tokens generated on one server will not work on any other server.
```
[the remark this answers is not in the document]
Comment (ericenns) [reply UPGRADING.md]: Fixed in e8014cd

**Review by apetkau — APPROVED** [review]
Looks great 👍 . Thanks

**Pull request merged**